json-rpc/src/runtime.rs

```diff
@@ -57,6 +63,15 @@ struct RpcResponseLog<'a> {
     response: &'a JsonRpcResponse,
 }
 
+// HealthCheckParams is optional params for different layer's health check.
+// If no param is provided, server return 200 by default to indicate HTTP server is running health.
+#[derive(serde::Deserialize)]
+struct HealthCheckParams {
+    // Health check returns 200 when this param is provided and meet the following condition:
+    //   server latest ledger info timestamp >= server current time timestamp + duration_secs
+    pub duration_secs: Option<u64>,
```

Comment:
I think this should be set by a query param on the healthcheck (i.e. configured in the load balancer or whatever is doing the healthcheck). A couple reasons:

* There are multiple checks being done: load balancer, Kubernetes liveness probe, Kubernetes readiness probe. These are looking for different things (e.g. Kubernetes liveness is just "is it running", whereas Kubernetes readiness is "is it ready to serve traffic").
* It's usually faster to change the LB configuration if this needs to be tuned quickly (e.g. all your full nodes are lagging for some reason).

Reply:
it is a query param, see route `.and(warp::query().map(move |params: HealthCheckParams| params))`

Reply:
💯

Reply:
@xli, is my understanding correct that this would basically be
jsonrpc/-/healthy?duration_secs=60
if I wanted to make sure I was within a 60 second window?